docs/volumes.md

```diff
@@ -104,4 +104,59 @@ There is no limit on how much space an EmptyDir or HostPath volume can consume,
 
 In the future, we expect that EmptyDir and HostPath volumes will be able to request a certain amount of space using a [resource](./resources.md) specification, and to select the type of media to use, for clusters that have several media types.
 
+# Mounters
+
+Before Kubernetes start containers in a pod, it mounts all volumes to a subdirectory in /var/lib/kubelet/pods/<pod UUID>. It shares the volumes to the container using `docker -v` parameter or similar mechanism.
+
+Kubernetes can mount the volumes in various ways, controlled by `kubelet --volume-mounter=XYZ` parameter.
+
+## Mounter
+`kubelet --volume-mounter=mount` (default) uses simple `mount -t <fstype> <what> <where>` call. All mount utilities (like `mount.nfs` or `mount.glusterfs`) must be present on the host.
+
+## NsenterMounter
+`kubelet --volume-mounter=nsenter` (default when kubelet runs inside a container, i.e. `kubelet --containerized=true` is used) should be used when `kubelet` itself runs in a container.
+
+It calls `nsenter --mount=/rootfs/proc/1/ns/mnt /usr/bin/mount -t <fstype> <what> <where>` from inside the container where `kubelet` runs. In other words, it enters the host namespace and calls `/usr/bin/mount` there. The mount utilities (like `mount.nfs` or `mount.glusterfs`) must be present on the **host**, not inisde `kubelet` container. Of course, `kubelet` container must be privileged in order to escape its own container and enter the host mount namespace.
+
+## ContainerMounter
+`kubelet --volume-mounter=container` assumes, that the host does not have some mount utilities (like `mount.glusterfs`) and uses container(s) with these utilities to mount certain filesystems. This mounter is configured by `--mount-container=xxx` option, which tells which container should be used for which filesystem. Complete syntax is:
```

Comment:
FYI, the propagation mode of bind-mounts in docker 1.7 was changed back to private, so you're going to be limited to docker 1.6 for this for now.